### PR TITLE
fix the L2 regularization of Adam and Adamax.

### LINKED ISSUE #5836: Fix the bug that regularization does not take effect in Adam
Adam works well in practice and compares favorably to other adaptive learning-method algorithms. It becomes a popular (almost a default optimizer for many tasks) optimizer for deep neural networks.

But the current implementation of Adam in V2 API ignore the `weight_decay_rate`  parameter. This means the `l2` regularization does not work for Adam and Adamax at all even if the user sets it.

A recent paper [Fixing Weight Decay Regularization in Adam](https://arxiv.org/abs/1711.05101) points out that decoupling weight decay and the optimization steps achieve a better learning performance.

Correctly implementing regularization is important for a learning task. 

We have this PR https://github.com/PaddlePaddle/Paddle/pull/2097 , but it does not correctly implement the L2 regularization in Adam and Adamax.

A related issue reported by one of our users: https://github.com/PaddlePaddle/Paddle/issues/4162

I will fix this.

### PR DESCRIPTION
**Pull request by lcy-seso**
fixes https://github.com/PaddlePaddle/Paddle/issues/5836 https://github.com/PaddlePaddle/Paddle/issues/4162

related to https://github.com/PaddlePaddle/Paddle/issues/5901 https://github.com/PaddlePaddle/Paddle/issues/5899

### CODE REVIEW

**Comment by luotao1** [comment]
感谢您给PaddlePaddle贡献代码。由于Paddle V1/V2版本已不再维护，相关代码也已从develop分支上删除，因此关闭您的PR，欢迎您向Paddle最新版-Fluid贡献代码。
Thanks for contributing to PaddlePaddle! Since V1/V2 will not be maintained anymore, and related codes have been deleted from develop branch as well, we close this PR. Welcome to contribute to Fluid——the latest version of PaddlePaddle.
